### PR TITLE
Add GFN-FF Implementation

### PR DESCRIPTION
**Pull request by sespic**
GFN-FF implementation for `xtb`, see https://onlinelibrary.wiley.com/doi/pdf/10.1002/anie.202004239 for details on the method.

### CODE REVIEW

File: src/gfnff/calculator.f90 [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+   efix = 0.0_wp
+
+   ! ------------------------------------------------------------------------
+   !  actual calculation
+   select case(mode_extrun)
+   case default
+      call gfnff_eg(gff_print,mol%n,ichrg,mol%at,mol%xyz,make_chrg,gradient,energy,results)
+   case(p_ext_gfnff)
+      call gfnff_eg(gff_print,mol%n,ichrg,mol%at,mol%xyz,make_chrg,gradient,energy,results)
+   end select
```
Comment (awvwgk) [inline src/gfnff/calculator.f90:127]: No need for the check on the `mode_extrun` here
```suggestion
   ! ------------------------------------------------------------------------
   !  actual calculation
   call gfnff_eg(gff_print,mol%n,ichrg,mol%at,mol%xyz,make_chrg,gradient,energy,results)
```

File: src/gfnff/d3com.f90 [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+! but WITHOUT ANY WARRANTY; without even the implied warranty of
+! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+! GNU Lesser General Public License for more details.
+!
+! You should have received a copy of the GNU Lesser General Public License
+! along with xtb.  If not, see <https://www.gnu.org/licenses/>.
+
+module gff_d3com
+      use iso_fortran_env, only : wp => real64
+      use xtb_disp_dftd4, only : r2r4 => r4r2, rcov
```
Comment (awvwgk) [inline src/gfnff/d3com.f90:20]: We should probably not forward this and just get rid of the module entirely.
`xtb_disp_dftd4` is already forwarding this from another module.

File: src/gfnff/dmat.f [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+! along with xtb.  If not, see <https://www.gnu.org/licenses/>.
+
+cccccccccccccccccccccccccccccccccccccccccccccc
+c density matrix
+c C: MO coefficient
+c X: scratch
+c P  dmat
+cccccccccccccccccccccccccccccccccccccccccccccc
+
+      subroutine dmat(ndim,focc,C,P)
```
Comment (awvwgk) [inline src/gfnff/dmat.f:25]: Can `dmat` from `xtb_scc_core` used instead?

File: src/gfnff/fermi.f [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+! GNU Lesser General Public License for more details.
+!
+! You should have received a copy of the GNU Lesser General Public License
+! along with xtb.  If not, see <https://www.gnu.org/licenses/>.
+
+ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+c fermi smearing
+ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+
+      SUBROUTINE FERMISMEAR(prt,NORBS,NEL,T,eig,occ,fod,e_fermi,s)
```
Comment (awvwgk) [inline src/gfnff/fermi.f:22]: We have a `fermismear` implementation in `xtb_scc_core` we should use instead.

File: src/gfnff/gdisp0.f90 [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+! You should have received a copy of the GNU Lesser General Public License
+! along with xtb.  If not, see <https://www.gnu.org/licenses/>.
+
+
+!CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC
+!      The   N E W   gradC6 routine    C
+!CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC
+
+      subroutine getdC6gfnff(mxci,mxcj,cni,cnj,&
+     &                       izi,izj,iat,jat,c6,dc6i,dc6j)
```
Comment (awvwgk) [inline src/gfnff/gdisp0.f90:24]: Is there any user of this routine or can it be deleted?

File: src/gfnff/gdisp0.f90 [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+        dc6j=0.0d0
+      end if
+!      nenner=nenner+1.d-9   ! can not be smaller than 1.d-10 in highly coordinated systems like metal clusters for cnmax=6
+!      c6  =zaehler/nenner
+!      dc6i=((dzaehler_i*nenner)-(dnenner_i*zaehler)) /(nenner*nenner)
+!      dc6j=((dzaehler_j*nenner)-(dnenner_j*zaehler)) /(nenner*nenner)
+
+      end
+
+module gffmod_dftd3
```
Comment (awvwgk) [inline src/gfnff/gdisp0.f90:93]: We should match the general naming convention:
```suggestion
module xtb_gfnff_gdisp0
```

File: src/gfnff/gffcom.f90 [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+      data en/ 2.200,3.000,0.980,1.570,2.040,2.550,3.040,3.440,3.980 &
+     &        ,4.500,0.930,1.310,1.610,1.900,2.190,2.580,3.160,3.500 &
+     &        ,0.820,1.000,1.360,1.540,1.630,1.660,1.550,1.830,1.880 &
+     &        ,1.910,1.900,1.650,1.810,2.010,2.180,2.550,2.960,3.000 &
+     &        ,0.820,0.950,1.220,1.330,1.600,2.160,1.900,2.200,2.280 &
+     &        ,2.200,1.930,1.690,1.780,1.960,2.050,2.100,2.660,2.600 &
+     &,0.79,0.89,1.10,1.12,1.13,1.14,1.15,1.17,1.18,1.20,1.21,1.22 &
+     &,1.23,1.24,1.25,1.26,1.27,1.3,1.5,1.7,1.9,2.1,2.2,2.2,2.2 &   ! value of W-Au modified
+!org &,1.23,1.24,1.25,1.26,1.27,1.3,1.5,2.36,1.9,2.2,2.20,2.28,2.54 &
+     &,2.00,1.62,2.33,2.02,2.0,2.2,2.2/
```
Comment (awvwgk) [inline src/gfnff/gffcom.f90:97]: This should probably be `parameter`?

File: src/gfnff/gffcom.f90 [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+     &0.32D0,0.37D0,1.30D0,0.99D0,0.84D0,0.75D0,0.71D0,0.64D0,0.60D0,&
+     &0.62D0,1.60D0,1.40D0,1.24D0,1.14D0,1.09D0,1.04D0,1.00D0,1.01D0,&
+     &2.00D0,1.74D0,1.59D0,1.48D0,1.44D0,1.30D0,1.29D0,1.24D0,1.18D0,&
+     &1.17D0,1.22D0,1.20D0,1.23D0,1.20D0,1.20D0,1.18D0,1.17D0,1.16D0,&
+     &2.15D0,1.90D0,1.76D0,1.64D0,1.56D0,1.46D0,1.38D0,1.36D0,1.34D0,&
+     &1.30D0,1.36D0,1.40D0,1.42D0,1.40D0,1.40D0,1.37D0,1.36D0,1.36D0,&
+     &2.38D0,2.06D0,1.94D0,1.84D0,1.90D0,1.88D0,1.86D0,1.85D0,1.83D0,&
+     &1.82D0,1.81D0,1.80D0,1.79D0,1.77D0,1.77D0,1.78D0,1.74D0,1.64D0,&
+     &1.58D0,1.50D0,1.41D0,1.36D0,1.32D0,1.30D0,1.30D0,1.32D0,1.44D0,&
+     &1.45D0,1.50D0,1.42D0,1.48D0,1.46D0/
```
Comment (awvwgk) [inline src/gfnff/gffcom.f90:113]: Use `xtb_param_atomicrad` instead?

File: src/gfnff/gffcom.f90 [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+     &91.22000d0, 92.90640d0, 95.94000d0, 98.90620d0, 101.0700d0,&
+     &102.9055d0, 106.4000d0, 107.8680d0, 112.4100d0, 114.8200d0,&
+     &118.6900d0, 121.7500d0, 127.6000d0, 126.9045d0, 131.3000d0,&
+     &132.9054d0, 137.3300d0,&
+     &138.91,140.12,140.91,144.24,147.00,150.36,151.97,157.25,&
+     &158.93,162.50,164.93,167.26,168.93,173.04,174.97,&
+     &178.4900d0, 180.9479d0,&
+     &183.8500d0, 186.2070d0, 190.2000d0, 192.2200d0, 195.0900d0,&
+     &196.9665d0, 200.5900d0, 204.3700d0, 207.2000d0, 208.9804d0,&
+     &209.,210.,222.,21*0.000d0/
```
Comment (awvwgk) [inline src/gfnff/gffcom.f90:163]: Any reason to duplicate the masses from `xtb_param_atomicmass`?

File: src/gfnff/gfnff_eg.f90 [hunk 10]
```diff
@@ -0,0 +1,3218 @@
...
+subroutine abhgfnff_eg3(n,A,B,H,at,xyz,q,sqrab,srab,energy,gdr)
+      use gff_param,only:hbacut,hbscut,xhbas,rad,hbalp,hblongcut,hbsf,hbst,xhaci_coh,hbabmix,nb,repz,hbnbcut,tors_hb,bend_hb
+      implicit none
+      integer A,B,H,n,at(n)
+      real*8 xyz(3,n),energy,gdr(3,n)
+      real*8 q(n)
+      real*8 sqrab(n*(n+1)/2)   ! squared dist
+      real*8 srab(n*(n+1)/2)    ! dist
+
+      real*8 C,D
```
Comment (awvwgk) [inline src/gfnff/gfnff_eg.f90:2033]: Shouldn't this be integers?

File: src/gfnff/gfnff_eg.f90 [hunk 11]
```diff
@@ -0,0 +1,3218 @@
...
+            eext = eext + r3(1)*(xyz(1,i)-xyze0(1,i))+&
+     &                    r3(2)*(xyz(2,i)-xyze0(2,i))+&
+     &                    r3(3)*(xyz(3,i)-xyze0(3,i))
+         enddo
+      endif
+
+!!!!!!!!!!!!!!!!!!
+! total energy
+!!!!!!!!!!!!!!!!!!
+ 99   etot = ees + edisp + erep + ebond &
```
Comment (awvwgk) [inline src/gfnff/gfnff_eg.f90:520]: goto mark not needed here

File: src/gfnff/gfnff_eg.f90 [hunk 12]
```diff
@@ -0,0 +1,3218 @@
...
+         rmul2 =  deddt / (rcb2*rp)
+         dedc = dedc*rmul2
+         dedb = deda+dedc
+         term1(1:3)=ea*damp2ij*dampjk*vab(1:3)
+         term2(1:3)=ea*damp2jk*dampij*vcb(1:3)
+         g(1:3,1) = -dedb(1:3)*damp-term1(1:3)-term2(1:3)
+         g(1:3,2) =  deda(1:3)*damp+term1(1:3)
+         g(1:3,3) =  dedc(1:3)*damp+term2(1:3)
+
+      end
```
Comment (awvwgk) [inline src/gfnff/gfnff_eg.f90:842]: Please end all subroutines with `end subroutine <name>`

File: src/gfnff/gfnff_eg.f90 [hunk 13]
```diff
@@ -0,0 +1,3218 @@
...
+      real*8  va(3),vb(3),vc(3),cosa
+      real*8  dt,ea,dedb(3),dedc(3),rmul2,rmul1,deddt
+      real*8  term1(3),term2(3),rab2,vab(3),vcb(3),rp
+      real*8  rcb2,damp,dampij,damp2ij,dampjk,damp2jk
+      real*8  theta,pi,deda(3),vlen,vp(3),et,dij,c1
+      real*8  term3(3),x1sin,x1cos,e1,dphi1,vdc(3)
+      real*8  ddd(3),ddc(3),ddb(3),dda(3),rjl,phi
+      real*8  omega,rij,rijk,phi0,rkl,rjk,dampkl,damp2kl
+      real*8  dampjl,damp2jl,valijklff,rn
+      parameter (pi = 3.1415926535897932384626433832795029d0)
```
Comment (awvwgk) [inline src/gfnff/gfnff_eg.f90:924]: We have this one in `xtb_mctc_constants`

File: src/gfnff/gfnff_eg.f90 [hunk 14]
```diff
@@ -0,0 +1,3218 @@
...
+      real*8  c0,kijk,va(3),vb(3),vc(3),cosa
+      real*8  dt,ea,dedb(3),dedc(3),rmul2,rmul1,deddt
+      real*8  term1(3),term2(3),rab2,vab(3),vcb(3),rp
+      real*8  rcb2,damp,dampij,damp2ij,dampjk,damp2jk
+      real*8  theta,pi,deda(3),vlen,vp(3),et,dij,c1
+      real*8  term3(3),x1sin,x1cos,e1,dphi1,vdc(3)
+      real*8  ddd(3),ddc(3),ddb(3),dda(3),rjl,phi
+      real*8  omega,rij,rijk,phi0,rkl,rjk,dampkl,damp2kl
+      real*8  dampjl,damp2jl,valijklff,rn
+      parameter (pi = 3.1415926535897932384626433832795029d0)
```
Comment (awvwgk) [inline src/gfnff/gfnff_eg.f90:986]: Again, use `xtb_mctc_constants`

File: src/gfnff/gfnff_ini.f90 [hunk 15]
```diff
@@ -0,0 +1,1820 @@
...
+      real(wp) shift,dum,dum1,dum2,dum4,qafac,fqq,feta
+      real(wp) sumppi,fpi,fxh,fijk,fsrb2,ees
+      real(wp) fheavy,fn,eold,fctot,fij
+      real(wp) hbpi(2),hbpj(2),sdum3(3)
+      real(wp) bstrength
+      real(wp) xx(20)
+      real(wp) fkl,qreps,fbsmall,bohr
+
+      parameter (pi=3.1415926535897932384626433832795029_wp)
+      parameter (bohr=1.0_wp/0.52917726_wp)
```
Comment (awvwgk) [inline src/gfnff/gfnff_ini.f90:59]: Definitely needs to get replaced with `xtb_mctc_convert`

File: src/gfnff/gfnff_ini.f90 [hunk 16]
```diff
@@ -0,0 +1,1820 @@
...
+      real(wp) r0,pi,ff,omega,f1,f2,phi,valijklff,ringf,fcn
+      real(wp) shift,dum,dum1,dum2,dum4,qafac,fqq,feta
+      real(wp) sumppi,fpi,fxh,fijk,fsrb2,ees
+      real(wp) fheavy,fn,eold,fctot,fij
+      real(wp) hbpi(2),hbpj(2),sdum3(3)
+      real(wp) bstrength
+      real(wp) xx(20)
+      real(wp) fkl,qreps,fbsmall,bohr
+
+      parameter (pi=3.1415926535897932384626433832795029_wp)
```
Comment (awvwgk) [inline src/gfnff/gfnff_ini.f90:58]: Use value from `xtb_mctc_constants`

File: src/prog/main.F90 [hunk 17]
```diff
@@ -1168,9 +1227,15 @@ subroutine parseArguments(env, args, inputFile, paramFile, accuracy, lgrad, &
...
-         call set_exttyp('eht')    !ppracht 10/2018 GFN0
+         call set_exttyp('eht')
          call env%warning("The use of '"//flag//"' is discouraged, " //&
             & "please use '--gfn 0' next time", source)
+      
+      case('--gfnff')
+         call set_exttyp('ff')
+      
+      case('--gff')
+         call set_exttyp('ff')
```
Comment (awvwgk) [inline src/prog/main.F90:1238]: This flag is not documented

File: src/optimizer.f90 [hunk 18]
```diff
@@ -980,17 +980,24 @@ subroutine modhes(env, modh, natoms, xyz, chg, Hess, pr)
...
-     if (pr) write(env%unit,'(a)') "Using Lindh-Hessian"
+     if (pr) write(env%unit,'(x,a)') "Using Lindh-Hessian"
      call mh_lindh_d2(xyz, natoms, Hess, chg, modh)
    case(p_modh_lindh)
-     if (pr) write(env%unit,'(a)') "Using Lindh-Hessian (2007)"
+     if (pr) write(env%unit,'(x,a)') "Using Lindh-Hessian (2007)"
      call mh_lindh(xyz, natoms, Hess, chg, modh)
    case(p_modh_swart)
-     if (pr) write(env%unit,'(a)') "Using Swart-Hessian"
+     if (pr) write(env%unit,'(x,a)') "Using Swart-Hessian"
```
Comment (awvwgk) [inline src/optimizer.f90:992]: What is the motivation for this change? Also the `x` specifier without a width is an error with GNU compilers and debug settings.

File: src/optimizer.f90 [hunk 19]
```diff
@@ -980,17 +980,24 @@ subroutine modhes(env, modh, natoms, xyz, chg, Hess, pr)
...
      call mh_swart(xyz, natoms, Hess, chg, modh)
+   case(p_modh_gff)
+     if (mode_extrun.eq.p_ext_gfnff) then        
+        if (pr) write(env%unit,'(x,a)') "Using GFN-FF Lindh-Hessian"
+        call gff_ddvopt(xyz, natoms, HEss, chg, modh%s6)
+     else
+        if (pr) write(env%unit,'(x,a)') "This Lindh-Hessian is only compatible with GFN-FF"
```
Comment (awvwgk) [inline src/optimizer.f90:999]: Same here, `x` is not a valid format specifier

File: src/io/reader/ctfile.f90 [hunk 20]
```diff
@@ -92,8 +93,10 @@ subroutine readMoleculeMolfile(mol, unit, status, iomsg)
    call getline(unit, line, error)
    read(line, '(20x,a2)', iostat=error) sdf_dim
    if (error == 0 .and. (sdf_dim == '2D' .or. sdf_dim == '2d')) then
-      iomsg = "two dimensional structures are not a valid input for this program"
-      return
+      !iomsg = "two dimensional structures are not a valid input for this program"
+      iomsg = "two dimensional structures input recognized"
```
Comment (awvwgk) [inline src/io/reader/ctfile.f90:97]: Don't write to `iomsg` if there was no error.

File: src/hessian.F90 [hunk 21]
```diff
@@ -228,6 +228,60 @@ subroutine numhess( &
       !$ call mkl_set_num_threads(nproc)
 #endif
 
+   else if (mode_extrun.eq.p_ext_gfnff) then
```
Comment (awvwgk) [inline src/hessian.F90:231]: There is a frozen hessian case as well, see above.

File: src/gfnff/occu.f90 [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+!
+! xtb is distributed in the hope that it will be useful,
+! but WITHOUT ANY WARRANTY; without even the implied warranty of
+! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+! GNU Lesser General Public License for more details.
+!
+! You should have received a copy of the GNU Lesser General Public License
+! along with xtb.  If not, see <https://www.gnu.org/licenses/>.
+
+subroutine occu(ndim,nel,nopen,ihomoa,ihomob,focca,foccb)
```
Comment (awvwgk) [inline src/gfnff/occu.f90:18]: We have an implementation for this in `xtb_scc_core` as well.

File: src/gfnff/mdcom.f90 [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+!
+! xtb is distributed in the hope that it will be useful,
+! but WITHOUT ANY WARRANTY; without even the implied warranty of
+! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+! GNU Lesser General Public License for more details.
+!
+! You should have received a copy of the GNU Lesser General Public License
+! along with xtb.  If not, see <https://www.gnu.org/licenses/>.
+
+module gff_mdcom
```
Comment (awvwgk) [inline src/gfnff/mdcom.f90:18]: This module is not used anywhere -> delete.

File: src/gfnff/math.f [hunk 24]
```diff
@@ -0,0 +1,185 @@
...
+! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+! GNU Lesser General Public License for more details.
+!
+! You should have received a copy of the GNU Lesser General Public License
+! along with xtb.  If not, see <https://www.gnu.org/licenses/>.
+
+c  .....................................................................
+
+      subroutine vscal(a,n,scale)
+      implicit double precision (a-h,o-z), integer (i-n)
```
Comment (awvwgk) [inline src/gfnff/math.f:21]: Definitely a no-go. Use implicit none here.

File: src/gfnff/gfnff_setup.f90 [hunk 25]
```diff
@@ -0,0 +1,188 @@
...
+  if (allocated(mol%pdb)) then
+    read_file_type = 2
+    ini = .true.
+  else if (allocated(mol%sdf)) then
+    read_file_type = 1
+    ini = .false.
+  else
+    read_file_type = 0
+    ini = .true.
+  end if
```
Comment (awvwgk) [inline src/gfnff/gfnff_setup.f90:103]: We already have an enumerator this purpose in `xtb_mctc_filetypes` and `mol%ftype`.

File: src/hessian.F90 (hunk 21 above)
Comment (awvwgk) [reply src/hessian.F90:231]: Also a `select type` is certainly better for this purpose.

File: src/gfnff/dmat.f (hunk 3 above)
Comment (sespic) [reply src/gfnff/dmat.f:25]: yes, definitely!

File: src/gfnff/d3com.f90 (hunk 2 above)
Comment (sespic) [reply src/gfnff/d3com.f90:20]: It's gone.

File: src/gfnff/dmat.f (hunk 3 above)
Comment (sespic) [reply src/gfnff/dmat.f:25]: `xtb_scc_core` is now used

File: src/gfnff/gdisp0.f90 (hunk 5 above)
Comment (sespic) [reply src/gfnff/gdisp0.f90:24]: can be deleted

File: src/gfnff/gffcom.f90 (hunk 7 above)
Comment (sespic) [reply src/gfnff/gffcom.f90:97]: changed from `data` to `parameter`

File: src/gfnff/gffcom.f90 (hunk 9 above)
Comment (sespic) [reply src/gfnff/gffcom.f90:163]: now using `xtb_param_atomicmass`

File: src/gfnff/gfnff_eg.f90 (hunk 10 above)
Comment (sespic) [reply src/gfnff/gfnff_eg.f90:2033]: changed ti integers

File: src/gfnff/gfnff_eg.f90 (hunk 11 above)
Comment (sespic) [reply src/gfnff/gfnff_eg.f90:520]: erased

File: src/gfnff/gfnff_eg.f90 (hunk 13 above)
Comment (sespic) [reply src/gfnff/gfnff_eg.f90:924]: now using `xtb_mctc_constants`

File: src/gfnff/gfnff_eg.f90 (hunk 12 above)
Comment (sespic) [reply src/gfnff/gfnff_eg.f90:842]: done

File: src/gfnff/gfnff_eg.f90 (hunk 14 above)
Comment (sespic) [reply src/gfnff/gfnff_eg.f90:986]: now using `xtb_mctc_constants`

File: src/gfnff/gfnff_ini.f90 (hunk 15 above)
Comment (sespic) [reply src/gfnff/gfnff_ini.f90:59]: same here

File: src/gfnff/gfnff_ini.f90 (hunk 16 above)
Comment (sespic) [reply src/gfnff/gfnff_ini.f90:58]: done

File: src/prog/main.F90 (hunk 17 above)
Comment (sespic) [reply src/prog/main.F90:1244]: now documented

File: src/optimizer.f90 (hunk 18 above)
Comment (sespic) [reply src/optimizer.f90:992]: has been changed accordingly.

File: src/optimizer.f90 (hunk 19 above)
Comment (sespic) [reply src/optimizer.f90:999]: also changed back.

File: src/gfnff/occu.f90 (hunk 22 above)
Comment (sespic) [reply src/gfnff/occu.f90:18]: `occu` is now taken from ``xtb_scc_core

File: src/hessian.F90 (hunk 21 above)
Comment (sespic) [reply src/hessian.F90:231]: Now using `select type` and frozen hessian is added.

File: src/gfnff/mdcom.f90 (hunk 23 above)
Comment (sespic) [reply src/gfnff/mdcom.f90:18]: deleted

File: src/gfnff/math.f (hunk 24 above)
Comment (sespic) [reply src/gfnff/math.f:21]: Subroutine modified accordingly.

File: src/gfnff/gfnff_setup.f90 (hunk 25 above)
Comment (sespic) [reply src/gfnff/gfnff_setup.f90:103]: Now using `mol%ftype` and fileType from `xtb_mctc_filetypes`.

**Comment by awvwgk** [comment]
Travis CI errored with 429, too many requests. We got all three CI tests passing, but not for the same commit, unfortunately. Build with GCC8, GCC9, Intel18 and Intel19 both with CMake and meson were confirmed locally.
